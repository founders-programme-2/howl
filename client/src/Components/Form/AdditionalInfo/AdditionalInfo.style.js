import styled from 'styled-components';

const ErrMsg = styled.p`
  color: red;
  font-weight: bold;
`;

const ErrMsgInfoTrue = styled.p`
  color: red;
  font-weight: bold;
  margin: 2rem 2rem 0;
`;

export { ErrMsg, ErrMsgInfoTrue };
