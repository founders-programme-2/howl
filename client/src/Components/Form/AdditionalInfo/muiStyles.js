// TODO: Refactor media query to take in constant variable for size

export const styles = () => ({
  formControl: {
    display: 'block',
    marginLeft: '5%',
    marginRight: '5%',
    width: '50%',
    background: 'white',
    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
    borderRadius: 4,
    '@media (max-width: 768px)': {
      width: '80%',
    },
  },
  select: {
    width: '100%',
    paddingLeft: '2%',
  },
  formControlTags: {
    height: '50%',
  },
  textField: {
    marginLeft: '5%',
    marginRight: '5%',
    width: '50%',
    background: 'white',
    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
    borderRadius: 4,
    paddingLeft: '2%',
    '@media (max-width: 768px)': {
      width: '90%',
    },
  },
  label: {
    paddingLeft: '2%',
  },
  textFieldWlm: {
    width: '90%',
  },
  chips: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
    width: '97%',
    padding: 5,
    maxHeight: 100,
    overflowY: 'auto',
  },
  chip: {
    background: 'var(--highlight-color)',
    margin: '0.5%',
    color: 'white',
    fontWeight: 'bold',
    fontSize: 14,
  },
  formControlCheckbox: {
    display: 'block',
    width: '100%',
  },
  formControlLabel: {
    margin: '2%',
    paddingLeft: '2%',
    fontSize: 14,
  },
});

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
export const MenuProps = {
  PaperProps: {
    style: {
      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
      width: 250,
    },
  },
};
