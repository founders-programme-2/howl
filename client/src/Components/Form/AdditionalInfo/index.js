import React, { Fragment } from 'react';
import {
  withStyles,
  FormControl,
  Input,
  InputLabel,
  MenuItem,
  ListItemText,
  Select,
  Checkbox,
  Chip,
  FormControlLabel,
  TextField,
} from '../../muIndex';
import { styles, MenuProps } from './muiStyles';
import { ErrMsg, ErrMsgInfoTrue } from './AdditionalInfo.style';
import { tags as arrTags, categories as arrCategory } from '../../data.json';

const AdditionalInfo = ({
  classes,
  tags,
  tagsChange,
  category,
  dropdownChange,
  infoTrue,
  checkboxChange,
  textChange,
  wlmConnection,
  additionalComments,
  tagsErr,
  categoryErr,
  infoTrueErr,
  wlmConnectionErr,
  additionalCommentsErr,
}) => {
  return (
    <Fragment>
      {categoryErr ? (
        <ErrMsg>{categoryErr}</ErrMsg>
      ) : (
        <p>
          Please select one major category under which your contribution falls
          under. We understand that there might be more than one fits, please
          select one and you will be given a chance to select some subcategories
          later.
        </p>
      )}
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="category" className={classes.label}>
          Category (required)
        </InputLabel>
        <Select
          aria-label="category"
          value={category}
          onChange={dropdownChange}
          inputProps={{
            name: 'category',
            id: 'category',
          }}
          disableUnderline
          className={classes.select}
        >
          {arrCategory.sort().map(cat => (
            <MenuItem value={cat} key={cat}>
              {cat}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
      {tagsErr ? (
        <ErrMsg>{tagsErr}</ErrMsg>
      ) : (
        <p>
          Please select one or more &#39;tags&#39; for your contribution. A tag
          can be thought of as a related theme or subcategory.{' '}
        </p>
      )}
      <FormControl
        className={`${classes.formControl} ${classes.formControlTags}`}
      >
        <InputLabel htmlFor="selectMultipleCheckbox" className={classes.label}>
          Tags (required)
        </InputLabel>
        <Select
          aria-label="tags"
          multiple
          value={tags}
          onChange={tagsChange}
          input={<Input id="selectMultipleCheckbox" />}
          disableUnderline
          className={classes.select}
          renderValue={selected => (
            <div className={classes.chips}>
              {selected.map(value => (
                <Chip key={value} label={value} className={classes.chip} />
              ))}
            </div>
          )}
          MenuProps={MenuProps}
        >
          {arrTags.sort().map(tag => (
            <MenuItem key={tag} value={tag}>
              <Checkbox checked={tags.indexOf(tag) > -1} />
              <ListItemText primary={tag} />
            </MenuItem>
          ))}
        </Select>
      </FormControl>
      {wlmConnectionErr ? (
        <ErrMsg>{wlmConnectionErr}</ErrMsg>
      ) : (
        <p>
          Tell us about yourself. What is your connection to the Women&#39;s
          Liberation Movement? Please do not share any private information here
          such as your phone number or address.
        </p>
      )}
      <TextField
        id="standard-multiline-static"
        label="Connection to Womenâ€™s Liberation Movement (required)"
        name="wlmConnection"
        value={wlmConnection}
        multiline
        rows="4"
        fullWidth
        placeholder="examples: activist, daughter of an activist, et cetera)"
        className={`${classes.textField} ${classes.textFieldWlm}`}
        margin="normal"
        onChange={textChange('wlmConnection')}
        InputProps={{
          disableUnderline: true,
        }}
        InputLabelProps={{
          className: classes.label,
        }}
      />
      {additionalCommentsErr ? (
        <ErrMsg>{additionalComments}</ErrMsg>
      ) : (
        <p>Anything else you would like to tell us?</p>
      )}
      <TextField
        id="additionalComments"
        label="Additional comments? (optional)"
        name="additionalComments"
        value={additionalComments}
        multiline
        rows="4"
        placeholder="Talk to us!"
        className={classes.textField}
        margin="normal"
        onChange={textChange('additionalComments')}
        InputProps={{
          disableUnderline: true,
        }}
        InputLabelProps={{
          className: classes.label,
        }}
      />
      <FormControl className={classes.formControlCheckbox}>
        {infoTrueErr ? <ErrMsgInfoTrue>{infoTrueErr}</ErrMsgInfoTrue> : null}
        <FormControlLabel
          className={classes.formControlLabel}
          control={
            <Checkbox
              checked={infoTrue}
              onChange={checkboxChange('infoTrue')}
              value="infoTrue"
            />
          }
          label="I certify that this information is true to the best of my knowledge and contains no personal attacks."
        />
      </FormControl>
    </Fragment>
  );
};

export default withStyles(styles)(AdditionalInfo);
