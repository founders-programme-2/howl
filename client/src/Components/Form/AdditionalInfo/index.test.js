import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import AdditionalInfo from './index';

test('Natural', () => {
  let state = {};
  const props = {
    tags: [],
    category: '',
    infoTrue: false,
    wlmConnection: '',
    additionalComments: '',
    tagsChange: event => {
      state = { ...state, tags: event.target.value };
    },
    dropdownChange: event => {
      this.setState({ [event.target.name]: event.target.value });
    },
    checkboxChange: name => event => {
      state = { ...state, [name]: event.target.checked };
    },
    textChange: name => event => {
      state = { ...state, [name]: event.target.value };
    },
  };

  const additionalInfo = renderer
    .create(
      <BrowserRouter>
        {' '}
        <AdditionalInfo {...props} />{' '}
      </BrowserRouter>
    )
    .toJSON();
  expect(additionalInfo).toMatchSnapshot();
});
