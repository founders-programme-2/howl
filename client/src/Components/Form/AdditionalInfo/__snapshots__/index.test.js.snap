// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
Array [
  " ",
  <p>
    Please select one major category under which your contribution falls under. We understand that there might be more than one fits, please select one and you will be given a chance to select some subcategories later.
  </p>,
  <div
    className="MuiFormControl-root AdditionalInfo-formControl-1"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-5"
      data-shrink={false}
      htmlFor="category"
    >
      Category
    </label>
    <div
      aria-label="category"
      className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl AdditionalInfo-select-2"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="select-category"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="category"
          name="category"
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <p>
    Please select one or more 'tags' for your contribution. A tag can be thought of as a related theme or subcategory.
     
  </p>,
  <div
    className="MuiFormControl-root AdditionalInfo-formControl-1 AdditionalInfo-formControlTags-3"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-5"
      data-shrink={false}
      htmlFor="selectMultipleCheckbox"
    >
      Tags
    </label>
    <div
      aria-label="tags"
      className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl AdditionalInfo-select-2"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="selectMultipleCheckbox"
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <p>
    Tell us about yourself. What is your connection to the Women's Liberation Movement? Please do not share any private information here such as your phone number or address.
  </p>,
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root AdditionalInfo-textField-4 AdditionalInfo-textFieldWlm-6"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-5"
      data-shrink={false}
      htmlFor="standard-multiline-static"
    >
      Connection to Womenâ€™s Liberation Movement
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
      onClick={[Function]}
    >
      <textarea
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
        disabled={false}
        id="standard-multiline-static"
        name="wlmConnection"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="examples: activist, daughter of an activist, et cetera)"
        required={false}
        rows="4"
        value=""
      />
    </div>
  </div>,
  <p>
    Anything else you would like to tell us?
  </p>,
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root AdditionalInfo-textField-4"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated AdditionalInfo-label-5"
      data-shrink={false}
      htmlFor="additionalComments"
    >
      Additional comments?
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
      onClick={[Function]}
    >
      <textarea
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
        disabled={false}
        id="additionalComments"
        name="additionalComments"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Talk to us!"
        required={false}
        rows="4"
        value=""
      />
    </div>
  </div>,
  <div
    className="MuiFormControl-root AdditionalInfo-formControlCheckbox-9"
  >
    <label
      className="MuiFormControlLabel-root AdditionalInfo-formControlLabel-10"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-94 MuiCheckbox-root MuiCheckbox-colorSecondary"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <input
            checked={false}
            className="PrivateSwitchBase-input-97"
            data-indeterminate={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
            value="infoTrue"
          />
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
      >
        I certify that this information is true to the best of my knowledge and contains no personal attacks.
      </span>
    </label>
  </div>,
  " ",
]
`;
